README.md

```diff
@@ -9,67 +9,204 @@ Ceph-related Docker files.
+declares what ceph-container source to use, what architecture to build for, and what container image
+to use as the base for the build. See `make help` for a full description.
+
+#### Building images from staging
+It is possible to build container images directly from staging in the event that `make build` is not
+appealing. Simply stage the flavor(s) to be built, and then execute the desired build command for
+`daemon-base` and `daemon`.
+```
+# Example
+cd <staging>  
+docker build -t <daemon base tag> daemon-base/
+docker build -t <daemon tag> daemon/
+```
+
+## Contribution workflow
+The goal when adding contributions should be to make modifications or additions to the least
+specific files in order to reuse as much code as possible. Only specify specific changes when they
+absolutely apply only to the specific flavor(s) and not to others.
+
+### Fixing a bug
+1. Stage the flavor on which the bug was found (`make FLAVORS_TO_BUILD=<bugged build> stage`).
+2. Use the `find-src` script or `files-sources` list to locate the bugged file's source location.
+3. Edit the source location to fix the bug.
+4. Build test versions of the images (`make FLAVORS_TO_BUILD=<bugged build> build`).
+5. Test the images you built in your environment.
+6. Make a PR of your changes.
+
+### Adding a feature
+1. Determine the scope of the feature
+   - To which Ceph versions should the feature be added?
+   - To which distros should the feature be added?
+   - As a general guideline, new features should usually be added to all Ceph versions and distros.
+2. Add relevant changes to files in the project structure such that they will be added to only the
+   Ceph versions and distros in scope for the feature.
+3. Build test versions of the images (`make FLAVORS_TO_BUILD=<in-scope-flavors> build`).
+4. Test the images in your environment.
+5. Make a PR of your changes.
+
+### Adding a Ceph release
+Ideally, adding a new Ceph release is fairly easy. In the best case, all that needs done is adding
+flavors for the new Ceph version to the Makefile. At minimum, `ALL_BUILDABLE_FLAVORS` must be
+updated in the Makefile.
+
+In the worst case, trying to make as few modifications as possible:
+1. Add flavors for new Ceph versions to the Makefile.
+   - At minimum: `ALL_BUILDABLE_FLAVORS`.
+2. Edit `src/` files to support the new version if necessary, making sure not to break previous
+   versions.
+3. Edit `src/ALL/<distro>` files to support the new version if necessary, making sure not to break
+   previous versions.
+4. Add `src/<new ceph version>` files to support the new version if necessary.
+5. Build test versions of the images (`make FLAVORS_TO_BUILD=<new release flavors> build`).
+6. Test the images in your environment.
+7. Make a PR of your changes.
+
+### Adding a distro build
+1. Add flavors for the new distro to the Makefile.
+   - At minimum: `ALL_BUILDABLE_FLAVORS`.
+2. Add a `src/ALL/<new distro>` directory for the new distro
```

Comment:
The actual code doesn't do that at all as everything which is distro specific is in ceph-release/.
I'm currently updating the doc in my centos branch, I can handle all those changes if you like.

Reply:
Typo. Good catch. I'm working on patching the docs